nixos/modules/services/networking/xray.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  options = {
+
+    services.xray = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Whether to run xray server.
+
+          Either `configFile` or `config` must be specified.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.xray;
+        defaultText = literalExpression "pkgs.xray";
+        description = lib.mdDoc ''
+          Which xray package to use.
+        '';
+      };
+
+      configFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/xray/config.json";
+        description = lib.mdDoc ''
+          The absolute path to the configuration file.
+
+          Either `configFile` or `config` must be specified.
+
+          See <https://www.v2fly.org/en_US/config/overview.html>.
+        '';
+      };
+
+      config = mkOption {
```

Comment:
By convention we call this `settings` in NixOS.

Reply:
I don't want to break the options from v2ray since the config files are interchangeable between the programs (you may want to just change v2 to x when switching to xray)

https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/v2ray.nix

Reply:
In that case do you need a new service then? Just set the package on the `v2ray` module.

Also, I suggest we go back and cleanup `v2ray` module now to updated NixOS standards.